test/tasty/Data/Aeson/CommitTest.hs

```diff
@@ -80,6 +79,13 @@ tests = do
       , Left "Error in $.foo: parsing Int failed, expected Number, but encountered String"
       )
     ]
+  testParserWithCases
+    (runCommit . tryParser . objWithKey "bar" (objWithKey "foo" (parseJSON :: Value -> Parser Int)))
+    [ ("subparser retains path from super-parser"
+      , [aesonQQ| { "bar": {"foo": "fail"} } |]
+      , Left "Error in $.bar.foo: parsing Int failed, expected Number, but encountered String"
+      )
+    ]
```

Comment:
I'm having trouble understanding the rationale for this test. It doesn't look like it tests anything not already implicitly tested by "fails with nested relative path" above, and the test name seems to be copied from the one above it without testing anything resembling that name.

Reply:
This verifies that `tryParser` doesn't throw away the path. The heading should probably be updated, but I want to have a test to make sure we're not using the wrong paths in any of our combinators provided by the library.

Reply:
I've updated the name of the test now

Reply:
Thanks for updating the name.

I see. Having the test is fine. I don't think `tryParser` would ever have had a problem with `>>=` since loss of the path happens when a path is generated in the left side of `>>=` then needed for an error on the right side. `tryParser` trivially succeeds on the right side of `>>=` so it doesn't use the path even if it's lost, and a failing `tryParser` means failing on the left side where `>>=` never gets a chance to lose the path.